### ISSUE TITLE
marketmaker issue with libcurl.so.4: version `CURL_OPENSSL_3'  in Manjaro/Arch Distros

### ISSUE THREAD

**Issue by xenfasa** [KomodoPlatform/BarterDEX#196]
I'm no expert but after chatting with others in support we decided best to sumbit an issue here.
BarterDex stops at "Pleas wait.."  Apparently because marketmaker never starts Running.

Running marketmaker alone gives this

./marketmaker: /usr/lib/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by ./marketmaker)

Whether SSL3 is needed or not for marketmaker it does not start because of this version check.

Arch and Manjaro distros no longer include SSL3 as of this commit here:
https://git.archlinux.org/svntogit/packages.git/commit/trunk?h=packages/openssl&id=4b82ed4285c7cb76caf6d75a015c5a7542c625d1

It's because SSL3 on the way out
https://www.edgehosting.com/blog/2016/01/alternatives-for-tlsv1-and-sslv3-encryption/

So no surprise, with Arch/Manjaro being on the bleeding edge of things... (as some claim) has removed any potential dependency on SSL3

But apparently something in marketmaker's version checks it wants to see some SSL3 love but finds none and will never unless you do something like this (recompiling SSL3 in) which seems useless other than passing a version check.

http://vaughanhilts.me/blog/2017/02/06/openssl-with-sslv3-on-arch-linux-for-dot-net-core.html

I'm not sure what category this goes, so admin, please move this issue where if it belongs.

Cheers :)

**Comment by satindergrewal** [comment]
This is not exactly a UI repo issue, but instead a core `marketmaker` binary issue, which provides DEX API to this BarterDEX GUI.
I think it's better to be reported at either of these repos:

http://github.com/artemii235/SuperNET
or
http://github.com/jl777/SuperNET

Thanks

**Comment by artemii235** [comment]
@xenfasa Hi! Thank you for report. Seems like we should build separate mm version for Arch/Manjaro. Also, there is a workaround described in this issue: https://github.com/citra-emu/citra/issues/2976.
To fix it you should `sudo pacman -S libcurl-compat` and then `LD_PRELOAD=/usr/lib/libcurl.so.3 ./run_mm`. I will try to create separate Arch build on this week.
@satindergrewal How do you think, is it possible to run different binary from GUI depending on Linux distro?

**Comment by satindergrewal** [comment]
@artemii235 yes, it must be possible, I guess. The most easiest way to go with other arch specific builds done. Like if it's not for debian/ubuntu/redhat/centos specific and it's specially made for Arch/Majaro, then mention that as Arch build, and have the linux `marketmaker` replaced in that bundle. That means there'll be 2 Linux version of the bundled binaries provided to keep things simpler.
CC @himu007 @siulynot

**Comment by ShapeShifter499** [comment]
@artemii235 any updates on this?

**Comment by artemii235** [comment]
@ShapeShifter499 Hi, excuse me, I didn't have a chance to look at it yet. I will post a comment here once I fix it.
